### PR TITLE
Simplify options

### PR DESCRIPTION
**Pull request by amirebrahimi**
**Describe the changes**
Consolidated settings/options in MeshSimplifier to use SimplificationOptions

**How to test**
You can set any of the individual properties as normal or can now set the SimplificationOptions as a block.

**Screenshots**
This was done in order to enable easy exposure of the options in AutoLOD:
image

### CODE REVIEW

File: Runtime/MeshSimplifier.cs [hunk 1]
```diff
@@ -67,17 +67,9 @@ public sealed class MeshSimplifier
...
 
         #region Fields
-        private bool preserveBorderEdges = false;
-        private bool preserveUVSeamEdges = false;
-        private bool preserveUVFoldoverEdges = false;
-        private bool preserveSurfaceCurvature = false;
-        private bool enableSmartLink = true;
-        private int maxIterationCount = 100;
-        private double agressiveness = 7.0;
+        SimplificationOptions simplificationOptions = SimplificationOptions.Default;
```
Comment (Whinarn) [inline Runtime/MeshSimplifier.cs:70]: Normally I would have requested that the private modifier should be added here, to keep the consistency with the rest of the code. I know that it's optional, but personally I always prefer to be more explicit.
But I'll approve this anyways.

Comment (amirebrahimi) [reply Runtime/MeshSimplifier.cs:70]: Missed that one. We leave out C# defaults w/ our code conventions here at Unity, but I usually try to match where I am contributing to.

**Comment by amirebrahimi** [comment]
Thanks. Once I noticed your serializable struct it made it quick to support per-simplifier options, which has been a feature request (Unity-Technologies/AutoLOD#7) for some time.

Also, great work on the auto-CI for releases.

**Comment by Whinarn** [comment]
> Thanks. Once I noticed your serializable struct it made it quick to support per-simplifier options, which has been a feature request ([Unity-Technologies/AutoLOD#7](https://github.com/Unity-Technologies/AutoLOD/issues/7)) for some time.

Ah, cool.

> Also, great work on the auto-CI for releases.

I have just started using [semantic-release](https://github.com/semantic-release/semantic-release), and I'm still getting adjusted to it. I actually failed twice with this release. First because I totally missed that refactor didn't automatically trigger a release (which makes sense), and then it triggered a release for my revert of that temporary release fix 🤦
Anyways, I'm learning 😄

**Pull request merged**